src/public/components/Header/Header.js

```diff
@@ -0,0 +1,89 @@
+import AjaxModule from '../../modules/ajax.js';
+import ButtonComponent from '../Button/Button.js';
+
+export default class HeaderComponent {
+    constructor ({
+        parent = document.body,
+        pages = {}
+    } = {}) {
+        this._parent = parent;
+        this.pages = pages;
+        this.buttons = {};
+        this._elements = {};
+        this._elem = null;
+        this.namesAndTitles = null;
+        this.onClickItem = this.onClickItem.bind(this);
+    }
+
+    onClickItem (event) {
+        const currentTarget = event.target;
+
+        if ((currentTarget instanceof HTMLButtonElement)) {
```

Comment:
Давай уменьшим уровень вложенности и напишем
```suggestion
        if (!(currentTarget instanceof HTMLButtonElement)) {
            return;
        }
```

а дальше всё остальное.